.devcontainer/Dockerfile

```diff
@@ -0,0 +1,50 @@
+FROM rust:1.50
+
+# versions (according to their git tag) of different tools to download
+# https://github.com/EmbarkStudios/cargo-deny/releases
+ARG CARGO_DENY_VER=0.9.0
+# https://github.com/rust-lang/mdBook/releases
+ARG MDBOOK_VER=v0.4.7
+# https://github.com/rustwasm/wasm-pack/releases
+ARG WASM_PACK_VER=v0.8.1
+# https://nodejs.org/
+ARG NODE_MAJOR_VER=12
+
+ARG USER=pewpew
+
+# create the user
+RUN useradd -m -s /bin/bash -r -U $USER
+USER $USER
+WORKDIR /home/$USER
+
+# install rustfmt and clippy
+RUN rustup component add rustfmt clippy
+
+# install cargo-deny
+RUN curl -sSL https://github.com/EmbarkStudios/cargo-deny/releases/download/$CARGO_DENY_VER/cargo-deny-$CARGO_DENY_VER-x86_64-unknown-linux-musl.tar.gz \
+    | tar -xz --strip-components=1 -C $CARGO_HOME/bin --no-anchored cargo-deny
+
+# install mdbook
+RUN curl -sSL https://github.com/rust-lang/mdBook/releases/download/$MDBOOK_VER/mdbook-$MDBOOK_VER-x86_64-unknown-linux-gnu.tar.gz \
+    | tar -xz -C $CARGO_HOME/bin
+
+# install wasm-pack
+RUN curl -sSL https://github.com/rustwasm/wasm-pack/releases/download/$WASM_PACK_VER/wasm-pack-$WASM_PACK_VER-x86_64-unknown-linux-musl.tar.gz \
+    | tar -xz --strip-components=1 -C $CARGO_HOME/bin --no-anchored wasm-pack
+
+# install node
+RUN mkdir .node \
+    && SHALINE=$(curl -sSL https://nodejs.org/download/release/latest-v$NODE_MAJOR_VER.x/SHASUMS256.txt \
+        | grep "linux-x64.tar.xz") \
+    && FILENAME=$(echo $SHALINE | cut -d ' ' -f2) \
+    && curl -sSL https://nodejs.org/download/release/latest-v$NODE_MAJOR_VER.x/$FILENAME > $FILENAME \
+    && echo "$SHALINE\n" | shasum -sc - \
```

Comment:
Testing this locally (Ubuntu, not Docker), this command fails with the `\n` in the echo, but succeeds without it. Is there a difference running in docker?

Reply:
Yeah, it took me awhile to figure out. By default it seems like this Dockerfile is built with `/bin/sh` whereas by default in Ubuntu you're using `/bin/bash`. There were several things that worked in bash but not a standard shell. Changing the `echo` to `echo -e [...]` allows it to work in bash, but then in shell it will print out "-e [...]".